contentprovider.go

```diff
@@ -319,13 +320,28 @@ func matchScore(secs []DocumentSection, m *LineMatch) float64 {
 			}
 		}
 
+		if f.SymbolInfo != nil {
+			score += scoreKind(language, f.SymbolInfo.Kind)
+		}
+
 		if score > maxScore {
 			maxScore = score
 		}
 	}
 	return maxScore
 }
 
+func scoreKind(language string, kind string) float64 {
```

Comment:
This looks like the correct approach to me. Maybe mention you can do `universal-ctags --list-kinds=Java` to find out the mappings for a language.